tmt/base.py

```diff
@@ -2908,6 +2908,28 @@ def to_serialized(self) -> Dict[str, Any]:
     def from_serialized(cls, serialized: Dict[str, Any]) -> 'Result':
         return cls(serialized, name=serialized.pop('name'))
 
+    @staticmethod
+    def failures(log: Optional[str], msg_type: str = 'FAIL') -> str:
+        """ Filter stdout and get only messages with certain type """
+        if not log:
+            return ''
+        filtered = ''
+
+        # Filter beakerlib style logs
+        for m in re.findall(
+                fr'(^.*\[\s*{msg_type}\s*\][\S\s]*?)(?:^::\s+\[[0-9: ]+|:{{80}})',
```

Comment:
Just wonder: How exactly does the `[\S\s]` pattern work here?

Reply:
it's like '.' but this is matching also new lines

so this gets any line that has a FAIL (or other) beakerlib msg and then any other line until there is a line with the other match (:::::: or :: [time) which it does not save

Reply:
Thanks for clarification.